library_test.py

```diff
@@ -49,26 +54,23 @@ async def main() -> None:
     else:
         api = VodafoneStationSercommApi(args.router, args.username, args.password)
 
-    logged = False
-    exc = False
+    exc = True
     try:
```

Comment:
Applied, but doesn't seems to work here.
Debug shows `except VodafoneError:` block is not executed :disappointed:

Reply:
You are not raising in your exception handlers, the exception must go further out. So if you want to convert it to config flow auth exception,

You do a raise ConfigEntryAuthException() from exc, otherwise you should just log and do a `raise` so it propagate out.